content/docs/user-guide/programming/autogen/_index.md

```diff
@@ -1,10 +1,237 @@
 ---
-description: Use the Open 3D Engine Automated Generator to generate resources from Jinja2 templates. 
-linktitle: AZ Generator
-title: Automating boilerplate with AZ Generator
+description: Use the Open 3D Engine (O3DE) Automated Generator to generate resources from Jinja2 templates. 
+linktitle: AzAutoGen source generator
+title: Automate source generation from templates with AzAutoGen
 weight: 200
 ---
 
-{{< todo issue="https://github.com/o3de/o3de.org/issues/551" >}}
-Write documentation for the AZ Generator system.
-{{< /todo >}}
+When confronted with the problem of many similar code objects or source inputs, the best solution for simple cases is often automated generation from templates and values. **Open 3D Engine (O3DE)** has its own lightweight generator system, *AzAutoGen*, to solve this problem. AzAutoGen uses the [Jinja](https://jinja.palletsprojects.com/) template engine and Python in combination with either XML or JSON input files to produce generated sets of output. Your own O3DE Gems and projects can take advantage of AzAutoGen during [build](/docs/user-guide/build/generated-source/) to generate source inputs.
```

Comment:
"When confronted with the problem of many similar code objects or source inputs, [...]"

I don't feel like I understand the problem here. The word choice makes the problem domain too ambiguous. Can you rephrase, and follow with an example?

Reply:
Again, source files would probably read better than source inputs, as the generated files are outputs of the AzAutoGen system. Referring to them as source inputs could cause confusion.